src/core/TypedSTerm.ml

```diff
@@ -295,16 +295,18 @@ let const_of_cstor ?loc c = const ?loc c.cstor_id ~ty:c.cstor_ty
 
 let app_builtin ?loc ~ty b l =
   let mk_ b l = make_ ?loc ~ty (AppBuiltin(b,l)) in
-  begin match b, l with
-    | Builtin.Not, [f'] ->
-      begin match view f' with
+  begin match b, List.map view l with
+    | Builtin.Not, [_] ->
+      begin match view (List.hd l) with
```

Comment:
`Builtin.Not, [f'] -> begin match view f' with …` would be cleaner. `List.hd` is a code smell :(